### ISSUE TITLE
Packs depending on product stock don't auto update stock when product stock is updated.

### ISSUE THREAD

**Issue by therampagerado** [thirtybees/thirtybees#1593]
I have the following behaviour:

I have few packs of 5x and 10x products that rely on single products stock. When they got out of stock the packs also got out of stock.

When I updated the stock in my products the other day the packs did not auto update, even on save. I had to click between Calculate pack quantities automatically based on available quantities of items in pack and I want to specify available quantities manually so the stock updated correctly.

It was not a caching issue. I know that the implementation is pretty new but it would be nice to have some stock checker for the packs that runs every 10 or 15 minutes and updates their stock for such cases.

When a new pack is created the behaviour is correct - it picks the correct ammount right away.

**Comment by getdatakick** [comment]
Could you please provide more details about what you did exactly that packs did not recalculated? I can't reproduce the issue on my test environment -- dynamic packs are always calculated properly.

Note that there is already a data synchronization task, by default scheduled to run every 8 hours. You can adjust the frequency in `tb_scheduled_task` table.

**Comment by therampagerado** [comment]
Unfortunatley I did not wait 8 hours to check if it works.

As I said before, I had 2 packs created with the same product (5x and 10x). It ran out and when I restocked I went FO to check all caches and stuff but I saw that the packs are not updated.

Then I went BO and they showed 0. I had to manually click throu the 2 options above in order for the pack to get the new product availiablity and show the proper pack quantities.

Here are 
![image](https://user-images.githubusercontent.com/11251370/209579848-761c9d2f-d2ea-4b5d-894c-1742c1b342c4.png)
my pack settings:

**Comment by getdatakick** [comment]
Did you have to do that only once, or does this happen every time the quantities of pack items changed?

How do you change the quantities of pack items? Manually through back office, or through some import process?

**Comment by therampagerado** [comment]
This only happens when I run out of products for the pack and then manually add products in BO. When I sell individual products the pack quantities are corrected acordingly.

**Comment by therampagerado** [comment]
With the most current restock the packs updated instantly. Closing this issue.
